src/test/scala/chiselTests/LayerSpec.scala

```diff
@@ -66,6 +68,21 @@ class LayerSpec extends ChiselFlatSpec with Utils with MatchesAndOmits {
     )()
   }
 
+  they should "be enabled with a trait" in {
+
+    class Foo extends RawModule {
+      layer.enable(A.B)
+      layer.enable(C)
+      // This should be a no-op.
+      layer.enable(layer.Layer.root)
```

Comment:
What happens if you enable a layer multiple times?

Reply:
As Megan noticed, there's are made `.distinct` before going to FIRRTL. Currently, if a layer is enabled multiple times in CIRCT it has no additional effect (it's not disallowed). This should be ironed out on the FIRRTL spec side.